web/src/lib/components/ui/add-resource/add-resource.svelte

```diff
@@ -1,125 +1,159 @@
-  let selectedTagIds: string[] = tags ? tags.map((tag) => tag.id) : [];
+
   let selectedSubjects = (subjects || []).map((subject) => ({
     id: subject.id,
     selected: entitySubjectsIds.has(subject.id),
   }));
-  let saved: boolean = false;
+
   let gradesAvailable: boolean = (grades || []).length > 0;
 
+  let form = superForm(data, {
+    validators: zodClient(schema),
+    SPA: true,
+  });
+  let { form: formData, enhance, validateForm } = form;
+  let names = Object.keys($formData).slice(1);
+
   async function updateGrades(value: string) {
     try {
       grades = await getGrades({ sort: ['name,asc'] }, { search: value });
     } catch {
       error(400, { message: 'Could not fetch grades' });
     }
   }
+
+  async function handleSubmit() {
+    let validatedForm = await validateForm();
+    if (!validatedForm.valid) {
+      return;
+    }
+    await update(
+      $formData,
+      selectedSubjects.filter((subject) => subject.selected).map((subject) => subject.id),
+    );
+    await goto('./');
+  }
 </script>
 
-<div class="grid grid-cols-[max-content,1fr] gap-8 p-4">
-  {#each descriptions as description, i}
-    <div class="text-lg font-bold flex">{description}</div>
-    <div class="flex flex-col gap-1 w-80">
-      <Input bind:value={values[i]} background={true} class="rounded-none font-normal flex max-w-80" />
-      {#if saved && !values[i].trim()}
-        <p class="text-sm text-red-600">Dieses Feld darf nicht leer sein.</p>
-      {/if}
-    </div>
-  {/each}
+<form method="POST" use:enhance>
+  <div class="grid grid-cols-[max-content,1fr] gap-8 p-4">
+    {#each descriptions as description, i}
+      <div class="text-lg font-bold flex">{description}</div>
+      <div class="flex flex-col gap-1 w-80">
+        <Form.Field {form} name={names[i]}>
+          <Form.Control let:attrs>
+            <Input
+              {...attrs}
+              bind:value={$formData[names[i]]}
+              background={true}
+              class="rounded-none font-normal flex max-w-80"
+            />
+          </Form.Control>
+          <Form.FieldErrors />
+        </Form.Field>
+      </div>
+    {/each}
 
-  {#if grades}
-    <div class="flex text-lg font-bold">Stufe:</div>
-    {#if gradesAvailable}
-      <div class="flex flex-col">
-        <ComboBox
-          onSearch={(value) => updateGrades(value)}
-          data={grades.map((grade) => ({ value: grade.id, label: grade.name }))}
-          bind:selectedId={entityGradeId}
-          shadow={true}
-        />
-        {#if saved && !entityGradeId}
-          <p class="text-sm text-red-600">Es muss eine Stufe ausgewählt werden.</p>
-        {/if}
+    {#if names.includes('capacity')}
+      <div class="text-lg font-bold flex">Kapazität:</div>
+      <div class="flex flex-col gap-1 w-80">
+        <Form.Field {form} name="capacity">
+          <Form.Control let:attrs>
+            <Input
+              type="number"
+              min="0"
+              {...attrs}
```

Comment:
This should be the first prop on the input, as you want everything else here to be overrides